src/sentry/api/endpoints/test_results.py

```diff
@@ -0,0 +1,198 @@
+            return super().to_representation(nodes)
+
+        except (KeyError, TypeError) as e:
+            # Handle malformed GraphQL response
+            sentry_sdk.capture_exception(e)
+            return []
+
+
+# Sample GraphQL response structure for reference
+sample_graphql_response = {
+    "data": {
+        "owner": {
+            "repository": {
+                "__typename": "Repository",
+                "testAnalytics": {
+                    "testResults": {
+                        "edges": [
+                            {
+                                "node": {
+                                    "updatedAt": "2025-05-22T16:21:18.763951+00:00",
+                                    "avgDuration": 0.04066228070175437,
+                                    "name": "../usr/local/lib/python3.13/site-packages/asgiref/sync.py::GetFinalYamlInteractorTest::test_when_commit_has_no_yaml",
+                                    "failureRate": 0.0,
+                                    "flakeRate": 0.0,
+                                    "commitsFailed": 0,
+                                    "totalFailCount": 0,
+                                    "totalFlakyFailCount": 0,
+                                    "totalSkipCount": 0,
+                                    "totalPassCount": 70,
+                                }
+                            },
+                            {
+                                "node": {
+                                    "updatedAt": "2025-05-22T16:21:18.763961+00:00",
+                                    "avgDuration": 0.034125877192982455,
+                                    "name": "../usr/local/lib/python3.13/site-packages/asgiref/sync.py::GetFinalYamlInteractorTest::test_when_commit_has_yaml",
+                                    "failureRate": 0.0,
+                                    "flakeRate": 0.0,
+                                    "commitsFailed": 0,
+                                    "totalFailCount": 0,
+                                    "totalFlakyFailCount": 0,
+                                    "totalSkipCount": 0,
+                                    "totalPassCount": 70,
+                                }
+                            },
+                        ],
+                    }
+                },
+            }
+        }
+    }
+}
+
+
+@region_silo_endpoint
+class TestResultsEndpoint(CodecovEndpoint):
+    owner = ApiOwner.CODECOV
+    publish_status = {
+        "GET": ApiPublishStatus.PUBLIC,
+    }
+
+    # Disable pagination requirement for this endpoint
+    def has_pagination(self, response):
+        return True
+
+    def get(self, request: Request, owner: str, repository: str, commit: str) -> Response:
+        """Retrieves the list of test results for a given commit."""
+
+        variables = {
+            "owner": owner,
+            "repo": repository,
+            "commit": commit,
```

Comment:
We don't seem to be using this variable in the query.

Also we are missing the filters, ordering and pagination vars as well right?